dotenv/README.md

```diff
@@ -0,0 +1,171 @@
+# Dotenv handling for deno.
+
+## Usage
+
+Setup a `.env` file in the root of your project.
+
+```sh
+# .env
+GREETING=hello world
+```
+
+Then import the configuration using the `config` function.
+
+```ts
+// app.ts
+import { config } from "https://deno.land/std@$STD_VERSION/dotenv/mod.ts";
+
+console.log(config());
+```
+
+Then run your app.
+
+```
+> deno app.ts
+{ GREETING: "hello world" }
+```
+
+### Options
+
+- `path?: string`: Optional path to `.env` file. Defaults to `./.env`.
+- `export?: boolean`: Set to `true` to export all `.env` variables to the
+  current processes environment. Variables are then accessable via
+  `Deno.env.get(<key>)`. Defaults to `false` (individual variables can be
+  exported via `export` keyword, see below).
+- `safe?: boolean`: Set to `true` to ensure that all necessary environment
+  variables are defined after reading from `.env`. It will read `.env.example`
+  to get the list of needed variables.
+- `example?: string`: Optional path to `.env.example` file. Defaults to
+  `./.env.example`.
+- `allowEmptyValues?: boolean`: Set to `true` to allow required env variables to
+  be empty. Otherwise it will throw an error if any variable is empty. Defaults
+  to `false`.
+- `defaults?: string`: Optional path to `.env.defaults` file which defaults to
+  `./.env.defaults`.
+
+### Auto loading
+
+`load.ts` automatically loads the local `.env` file on import and exports it to
+the process environment:
+
+```sh
+# .env
+GREETING=hello world
+```
+
+```ts
+// app.ts
+import "https://deno.land/std@$STD_VERSION/dotenv/load.ts";
+
+console.log(Deno.env.get("GREETING"));
+```
+
+```
+> deno --allow-env --allow-read app.ts
+hello world
+```
+
+### Safe Mode
+
+To enable safe mode, create a `.env.example` file in the root of the project.
+
+```sh
+# .env.example
+GREETING=
+export LOG_PATH="provide a value here"
```

Comment:
Is this correct? I've never seen `export` in dotenv file before. What's the reason for it?

Reply:
It's a convenience feature that was recently added by a contributor of my original library (https://github.com/pietvanzoen/deno-dotenv/pull/61).

It allows using a `.env` file via this lib and via `source .env`.

Reply:
Hmmmm.... I didn't realise it excludes non-exported variables. :/ that's not ideal.

Reply:
Ok... I've removed this functionality.